src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -371,16 +369,31 @@ public virtual bool TryInsert(TKey key, TValue value, out TValue? removed)
 
         private bool RemoveLast(out TValue? removed)
         {
-            TKey? key = _worstValue.GetValueOrDefault().Value;
+            TryAgain:
+            KeyValuePair<TValue, TKey> worstValue = _worstValue.GetValueOrDefault();
+            TKey? key = worstValue.Value;
             if (key is not null)
             {
-                return TryRemoveNonLocked(key, true, out removed, out _);
-            }
-            else
-            {
-                removed = default;
-                return false;
+                if (TryRemoveNonLocked(key, true, out removed, out _))
+                {
+                    return true;
+                }
+
+                TValue value = worstValue.Key;
+                if (_worstSortedValues.Remove(value))
+                {
+                    TGroupKey groupMapping = MapToGroup(value);
+                    if (_buckets.TryGetValue(groupMapping, out EnhancedSortedSet<TValue>? bucket))
+                    {
+                        UpdateSortedValues(bucket, previousLast: default);
+                    }
+                }
+
+                goto TryAgain;
```

Comment:
Is value removed from `_worstSortedValues` in all paths and `UpdateWorstValue()` called?

e.g. if `_buckets.TryGetValue` it will loop with the same value in `_worstValue.GetValueOrDefault()` and end up in infinite loop?